src/toml.rs

```diff
@@ -2,27 +2,62 @@ use crate::context;
 
 use std::{path, path::Path};
 
-use anyhow::{bail, Context as _};
+use anyhow::{bail, Context};
 use cargo_util_schemas::core::GitReference;
 use fs_err as fs;
 use pathdiff::diff_paths;
 
+pub enum Operation<'a> {
+    Add {
+        registry: &'a str,
+        name: &'a str,
+        mode: &'a context::Mode,
+    },
+    Remove {
+        name: &'a str,
+    },
+}
+
 pub fn patch_manifest(
     working_dir: &Path,
     manifest: &str,
     manifest_directory: &Path,
-    name: &str,
-    registry: &str,
-    mode: &context::Mode,
+    op: Operation,
```

Comment:
nit: I'm not sure we need this `Operation` type. I think this could be better off with this function removed, and just using `add_patch_to_manifest` and `remove_patch_from_manifest` directly, even if the below block of code needs to be duplicated and moved into those functions

```rust
let mut manifest: toml_edit::DocumentMut = manifest
        .parse()
        .context("patch manifest contains invalid toml")?;

    let manifest_table = manifest.as_table_mut();
```

Reply:
Do you have something in mind why to remove the single interface? I thought of using a struct, maybe named `Manifest`, which exposes the functions `add_override`, `remove_override` and perhaps later `list_overrides`.

Reply:
I was thinking of just having a two separate independent free functions, but a struct would also work great :+1:

Reply:
That being said this is a small thing so it's up to you if you'd like to change it